pkg/monitors/telegraf/monitors/dns/metadata.yaml

```diff
@@ -0,0 +1,33 @@
+monitors:
+- doc: |
+    This is an embedded form of the [Telegraf DNS Query
+    plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/dns_query).
+
+  dimensions:
+    plugin:
+      description: The plugin name "telegraf/dns".
+    domain:
+      description: The DNS domain queried.
+    server:
+      description: The DNS server queried.
+    record_type:
+      description: The DNS record type queried.
+    result:
+      description: Resolution result string (success = 0, timeout = 1, error = 2).
+    rcode:
+      description: Error string. See [this tab](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/dns_query#rcode-descriptions) for detailled descriptions.
+  metrics:
+    dns.query_time_ms:
+      description: Reponse time of the query in ms.
+      default: true
+      type: gauge
+    dns.result_code:
+      description: Resolution result code (success = 0, timeout = 1, error = 2).
+      default: true
+      type: gauge
+    dns.rcode_value:
```

Comment:
This metric makes me wary because of the same things I was saying before about dimensions and properties.  Because of the `rcode` dimension, this will make a new timeseries for every distinct error that happens in every unique {server, domain, query type} set.

Can you just remove the `rcode` dimension for now (use `emitter.OmitTag("rcode")` I think is enough to do it)?

Make sure the same thing applies to other metrics, where there will be one unique continuous timeseries for each unique {dns server, domain, query type} set for each metric.  This will make alerting much easier overall.

Reply:
hello @keitwb 

no problem, I think to this myself during my test but I was not sure about a telegraf based monitor if we should keep original implementation or transform to respect signalfx best practices.

Now I know.

Reply:
Sorry for the delay, I did my changes and tested but forgot to push.

I dropped the `result` and `rcode` dimensions and also the `telegraf_type` one which is always = `untyped`.

I preserved the `record_type` dimension which could change depending of the type of request (`NS` by default). seems useful to separate 2 metrics from the same `server` and `domain` but a different record type in case of multi instantiation.

I rebased the branch, updated doc and tested again.

it this ok for you @keitwb ?